src/libraries/Actions.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+/// @notice Library to define different pool actions.
+library Actions {
+    // TODO can explore having these as an enum but i think constants might be cheaper
+
+    uint256 constant SWAP = 0x00;
+    uint256 constant INCREASE_LIQUIDITY = 0x01;
+    uint256 constant DECREASE_LIQUIDITY = 0x02;
+    uint256 constant DONATE = 0x03;
+    uint256 constant CLEAR_DELTA = 0x04;
```

Comment:
nit: prefer just `CLEAR`? SETTLE/TAKE don't include `_DELTA`

Also I do think CLOSE & CLOSE_PAIR are needed in posm

Reply:
So I think we should discuss... I'm also just wondering if we're going a little too general here to make it work for different use cases. Like generalization is good but now its less optimal for posm, and router bc it includes things both of those use cases shouldnt support. 

ie POSM wont need swap, donate & router wont need increase/decrease.